### PR TITLE
Bind benchmark builder to Python

### LINKED ISSUE #1038: [FR][Python] Bind Benchmark methods / builder pattern
**Is your feature request related to a problem? Please describe.**
Missing features in Python, such as the ability to use`->MeasureProcessCPUTime()`, `->UseRealTime()`, `->Unit(benchmark::kMillisecond)`...

**Describe the solution you'd like**
We currently have 
```python
@register(name="Some benchmark)
def sum_million(state):
  while state:
    sum(range(1_000_000))
```
I suggest extending the API in a similar fashion to the C++ one. So something like
```python
@register(name="Some benchmark)
    .measure_process_time()
    .use_real_time()
    .unit(benchmark.kMilisecond)
def sum_million(state):
  while state:
    sum(range(1_000_000))
```

**Describe alternatives you've considered**
Alternatively, we could use Python named parameters inside the `register` function
```python
@register(
    name="Some benchmark)
    measure_process_time=True,
    use_real_time=True,
    unit=benchmark.kMilisecond,
)
def sum_million(state):
  while state:
    sum(range(1_000_000))
```
One downside is that in C++, the extra arguments to `BENCHMARK_REGISTER` are passed to the benchmark function.
Doing so here may prevent us from doing this in the future.

**Additional context**

### PR DESCRIPTION
**Pull request by AntoinePrv**
The following PR Close #1038
 - Bind `TimeUnit` and `BigO`
 - Bind `internal::Benchmark`
 - Modify `RegisterBenchmark` to return the benchmark object
 - Bind builder methods from `internal::Benchmark`
 - Add a Python decorator `option` to use the `Benchmark` options.

Example
```python
@benchmark.register
@benchmark.option.range_multiplier(2)
@benchmark.option.range(1 << 10, 1 << 18)
@benchmark.option.complexity(benchmark.oN)
def computing_complexity(state):
    while state:
        sum(range(state.range(0)))
    state.complexity_n = state.range(0)
```

Todo
 - [x] This builds on #1037 so preferably merge the other one first
 - [x] ~the multi-threaded benchmark is currently segfaulting and needs to be fixed~

### CODE REVIEW

**Comment by AntoinePrv** [comment]
Ok, so indeed I cannot run Python in multiple thread without acquiring the GIL.
I tried to register the threads and acquire the GIL, but it now deadlocks.
Probably due to

> In a multithreaded test (benchmark invoked by multiple threads simultaneously), it is guaranteed that none of the threads will start until all have reached the start of the benchmark loop

Anyways, seems to be out of scope for this issue

**Comment by dominichamon** [comment]
We should probably update the `__version__` in __init__.py too with this change.

**Comment by AntoinePrv** [comment]
I rebased on `master` and bumped version to `0.2.0`. I also fixed an indentation bug that arose in the `setup.py`.

File: setup.py [hunk 1]
```diff
@@ -58,71 +62,81 @@ def run(self):
...
         ]
 
         if IS_WINDOWS:
             # Link with python*.lib.
             for library_dir in self.library_dirs:
-                bazel_argv.append('--linkopt=/LIBPATH:' + library_dir)
+                bazel_argv.append("--linkopt=/LIBPATH:" + library_dir)
 
-            self.spawn(bazel_argv)
+        self.spawn(bazel_argv)
```
Comment (dominichamon) [inline setup.py:93]: i fixed this same issue in #1041  :)

well spotted!

**Comment by dominichamon** [comment]
LGTM but i'll wait a while in case there's any issues raised.

**Pull request merged**